### PR TITLE
Add oauth state param

### PR DESCRIPTION
**Pull request by xmflsct**
Help preventing potential CSRF https://auth0.com/docs/secure/attack-protection/state-parameters. Also make it work for the clients that do check the `state` param.

### CODE REVIEW

File: backend/src/errors/index.ts [hunk 1]
```diff
@@ -34,6 +34,10 @@ export function clientUnknown(): Response {
 	return generateErrorResponse(`The client is unknown or invalid`, 403)
 }
 
+export function notFound(): Response {
```
Comment (xtuc) [inline backend/src/errors/index.ts:37]: Could you please remove the 404 change from this PR? I think it needs a change

File: functions/oauth/authorize.ts [hunk 2]
```diff
@@ -73,11 +76,11 @@ export async function handleRequestPost(
 	if (person === null) {
 		url.pathname = '/first-login'
 		url.searchParams.set('email', identity.email)
-		url.searchParams.set('redirect_uri', encodeURIComponent(redirect_uri + '?code=' + code))
+		url.searchParams.set('redirect_uri', encodeURIComponent(redirect_uri + '?code=' + code + state))
 		return URLsafeRedirect(url.toString())
 	}
 
-	return URLsafeRedirect(redirect_uri + '?code=' + code)
+	return URLsafeRedirect(redirect_uri + '?code=' + code + state)
```
Comment (dario-piotrowicz) [inline functions/oauth/authorize.ts:83]: we've got two `redirect_uri + '?code=' + code + state`, wouldn't it be better to define a variable on line 74 for it and reuse it in both places? 

like 
```
const redirectUriParam = `redirect_uri?code=${code}${state}`
```
?

File: backend/src/errors/index.ts (hunk 1 above)
Comment (xmflsct) [reply backend/src/errors/index.ts:37]: Yep sorry that was not intentional.

File: functions/oauth/authorize.ts (hunk 2 above)
Comment (xmflsct) [reply functions/oauth/authorize.ts]: Checked

**Pull request closed without merging**